packages/strapi-plugin-upload/admin/src/components/Card/index.js

```diff
@@ -28,15 +32,59 @@ const Card = ({
   url,
   withFileCaching,
 }) => {
+  const ref = useRef(null);
+  const [, drop] = useDrop({
+    accept: ItemTypes.MEDIA_CARD,
+    hover(item) {
+      if (!ref.current) {
+        return;
+      }
+      const dragIndex = item.index;
+      const hoverIndex = index;
+
+      // Don't replace items with themselves
+      if (dragIndex === hoverIndex) {
+        return;
+      }
+
+      // Directly move the asset for faster reorder
+      // It doing so makes lot of computations though
+      moveAsset(dragIndex, hoverIndex);
+      // Note: we're mutating the monitor item here!
+      // Generally it's better to avoid mutations,
+      // but it's good here for the sake of performance
+      // to avoid expensive index searches.
+      item.index = hoverIndex;
+    },
+  });
+
   const fileSize = formatBytes(size, 0);
   const fileType = mime || type;
 
+  const [{ isDragging }, drag, preview] = useDrag({
+    item: { type: ItemTypes.MEDIA_CARD, id, index, checked, url, fileType },
+    canDrag: () => isDraggable,
+    collect: monitor => ({
+      isDragging: monitor.isDragging(),
+    }),
+  });
+
+  // Remove the default preview when the item is being dragged
+  // The preview is handled by the DragLayer
+  useEffect(() => {
+    preview(getEmptyImage(), { captureDraggingState: true });
+  }, [preview]);
+
+  const opacity = isDragging ? 0.2 : 1;
+
   const handleClick = () => {
     onClick(id);
   };
 
+  drag(drop(ref));
```

Comment:
All the logic of this file will be implemented in each card even if cards are not draggable (In the BrowseAsset tab and in the Plugin). I think we should create a DraggableCard that will be used only in the SelectedAssets list.